From 9939d8d804d0ce423824836173c296daf4e13e99 Mon Sep 17 00:00:00 2001
From: Douglas Gilbert <dgilbert@interlog.com>
Date: Mon, 7 Feb 2022 16:02:32 +0000
Subject: [PATCH] rescan-scsi-bus.sh: Use "seq -s"

Since the commit [git: 8c86fe2; svn rev 815  "library: add to '.so' name;
rescan-scsi-bus: multiple patches to sync with Suse], we correctly use
quoting when we print $idsearch or $lunsearch. This causes a change in
the output of the script, because these number lists are generated by
the GNU seq utility, which uses "\n" as default separator, and the
quoting now causes the newlines to be preserved. Use "seq -s" to
generate space-separated lists instead.



git-svn-id: https://svn.bingwo.ca/repos/sg3_utils/trunk@936 6180dd3e-e324-4e3e-922d-17de1ae2f315
---
 ChangeLog                  |  4 +++-
 scripts/rescan-scsi-bus.sh | 12 ++++++------
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/scripts/rescan-scsi-bus.sh b/scripts/rescan-scsi-bus.sh
index e4eeef1..78544b1 100755
--- a/scripts/rescan-scsi-bus.sh
+++ b/scripts/rescan-scsi-bus.sh
@@ -716,7 +716,7 @@ expandlist ()
       result="$result $beg";
     else
       end=${first#*-}
-      result="$result $(seq $beg $end)"
+      result="$result $(seq -s ' ' $beg $end)"
     fi
     [ "$rest" = "$first" ] && rest=""
     first=${rest%%,*}
@@ -1200,7 +1200,7 @@ fi
 unsetcolor
 debug=0
 lunsearch=
-opt_idsearch=$(seq 0 7)
+opt_idsearch=$(seq -s ' ' 0 7)
 filter_ids=0
 opt_channelsearch=
 remove=
@@ -1227,13 +1227,13 @@ while [ ! -z "$opt" ] && [ -z "${opt##-*}" ] ; do
     f) flush=1 ;;
     i) lipreset=0 ;;
     I) shift; lipreset=$opt ;;
-    l) lunsearch=$(seq 0 7) ;;
-    L) lunsearch=$(seq 0 "$2"); shift ;;
+    l) lunsearch=$(seq -s ' ' 0 7) ;;
+    L) lunsearch=$(seq -s ' ' 0 "$2"); shift ;;
     m) mp_enable=1 ;;
     r) remove=1 ;;
     s) resize=1; mp_enable=1 ;;
     u) update=1 ;;
-    w) opt_idsearch=$(seq 0 15) ;;
+    w) opt_idsearch=$(seq -s ' ' 0 15) ;;
     -alltargets)  existing_targets=;;
     -attachpq3) scan_flags=$((scan_flags|0x1000000)) ;;
     -channels=*)  arg=${opt#-channels=};opt_channelsearch=$(expandlist "$arg") ;;
@@ -1257,7 +1257,7 @@ while [ ! -z "$opt" ] && [ -z "${opt##-*}" ] ; do
     -sparselun) scan_flags=$((scan_flags|0x40)) ;;
     -sync) sync=2 ;;
     -update) update=1;;
-    -wide) opt_idsearch=$(seq 0 15) ;;
+    -wide) opt_idsearch=$(seq -s ' ' 0 15) ;;
     *) echo "Unknown option -$opt !" ;;
   esac
   shift
